skflow/data_feeder.py

```diff
@@ -176,3 +175,76 @@ def _feed_dict_fn():
         return _feed_dict_fn
+
+
+class DaskDataFeeder(object):
+    """Data feeder for TF trainer that reads data from dask.Series.
+
+    Numpy arrays can be serialized to disk and it's possible to do random seeks into them.
+    DaskDataFeeder will remove requirement to have full dataset in the memory and still do
+    random seeks for sampling of batches.
+
+    Parameters:
+        X: iterator that returns for each element, returns features.
+        y: iterator that returns for each element, returns 1 or many classes /
+           regression values.
+        n_classes: indicator of how many classes the target has.
+        batch_size: Mini batch size to accumulate.
+
+    Attributes:
+        X: input features.
+        y: input target.
+        n_classes: number of classes.
+        batch_size: mini batch size to accumulate.
+        input_shape: shape of the input.
+        output_shape: shape of the output.
+        input_dtype: dtype of input.
+        output_dtype: dtype of output.
+    """
+    def __init__(self, X, y, n_classes, batch_size, random_state=None):
+        import dask.dataframe as dd
+        x_dtype = np.int64 if X.dtype == np.int64 else np.float32
+        self.X = check_array(X, ensure_2d=False,
+                             allow_nd=True, dtype=x_dtype)
+        self.y = check_array(y, ensure_2d=False, dtype=np.float32)
+        self.X = X
+        self.y = y
+        # save column names
+        self.X_columns = list(X.columns)
+        self.y_columns = list(y.columns)
+        # combine into a data frame
+        self.df = dd.multi.concat([X, y], axis=1)
+
+        self.n_classes = n_classes
+        self.batch_size = batch_size
+        X_shape = tuple([X.count().compute()])
+        y_shape = tuple([y.count().compute()])
+        self.input_shape, self.output_shape = _get_in_out_shape(
+            X_shape, y_shape, n_classes, batch_size)
+        self.input_dtype, self.output_dtype = self.X.dtype, self.y.dtype
+        if random_state is None:
+            self.random_state = np.random.RandomState(42)
+        else:
+            self.random_state = random_state
+
+    def get_feed_dict_fn(self, input_placeholder, output_placeholder):
+        """Returns a function, that will sample data and provide it to given
+        placeholders.
+
+        Args:
+            input_placeholder: tf.Placeholder for input features mini batch.
+            output_placeholder: tf.Placeholder for output targets.
+        Returns:
+            A function that when called samples a random subset of batch size
+            from X and y.
+        """
+        def _feed_dict_fn():
+            inp = np.zeros(self.input_shape, dtype=self.input_dtype)
+            out = np.zeros(self.output_shape, dtype=self.output_dtype)
+            # TODO: option for with/without replacement (dev version of dask)
+            sample = self.df.random(self.batch_size/float(list(self.X_shape)[0]),
```

Comment:
also this is more like `sample_fraction` that you can assign in the `__init__`, e.g.
`self.sample_fraction = float(batch_size) / X_shape[0]`
and then here
`sample = self.df.random(self.sample_fraction)`